tests/torchtune/dev/rl/workers/test_ref_actor.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import functools
+
+import time
+
+import pytest
+import ray
+import torch
+from omegaconf import OmegaConf
+from ray.util.queue import Queue
+from tensordict import NonTensorData
+from tests.test_utils import gen_log_file_name, gpu_test
+from torchrl.data import LazyStackStorage, RayReplayBuffer
+from torchtune.dev.rl.datatypes.trajectory import Trajectory
+from torchtune.dev.rl.workers.ref_actor import RefActor
+
+grpo_samples = 4
+max_generated_tokens = 32
+
+
+@ray.remote(num_cpus=1, num_gpus=0)
+class DummyTrainer:
+    def __init__(self, rollout_queue):
+        self.rollout_queue = rollout_queue
+
+    def train(self):
+        while True:
+            size = self.rollout_queue.qsize()
+            if size == 0:
+                # TODO: this is a very bad thing
```

Comment:
ouchie :D Why do we need this?

Reply:
Admittedly still a Ray noob, but we need some way to send the signal that the queue is done and the test can terminate. However, PostProcessingWorker doesn't really have that control flow, it instead sleeps indefinitely waiting for the queue to repopulate. This is what I came up with to check that we've processed everything in the queue and can terminate the test. However, the queue is already empty before the model processes the final trajectory, so I added some buffer time to ensure the final trajectory gets processed. There may be an obviously better way to do this, if so I am all ears.